packages/office-addin-telemetry/src/telemetryJsonData.ts

```diff
@@ -0,0 +1,107 @@
+    try {
+        if (fs.existsSync(jsonFilePath)) {
+            const telemetryJsonData = readTelemetryJsonData(jsonFilePath);
+            if (groupNameExists(telemetryJsonData, groupName)) {
+            telemetryJsonData.telemetryInstances[groupName][property] = value;
+            fs.writeFileSync(jsonFilePath, JSON.stringify((telemetryJsonData), null, 2));
+            }
+        }
+    } catch {
+        console.log(chalk.default.red("Could not modify property!\n Make sure the group name and file exists!"));
+    }
+}
+/**
+ * Reads data from the telemetry json config file
+ * @param jsonFilePath Optional path to the json config file
+ * @returns Parsed object from json file if it exists
+ */
+export function readTelemetryJsonData(jsonFilePath = telemetryJsonFilePath): any {
+    if (fs.existsSync(jsonFilePath)) {
+        const jsonData = fs.readFileSync(jsonFilePath, "utf8");
+        return JSON.parse(jsonData.toString());
+    }
+}
+/**
+ * Returns telemetry level of telemetry object
+ * @param groupName Group name to search for in the specified json data
+ * @param jsonFilePath Optional path to the json config file
+ */
+export function readTelemetryLevel(groupName: string, jsonFilePath = telemetryJsonFilePath): string {
+        const jsonData = readTelemetryJsonData(jsonFilePath);
+        return jsonData.telemetryInstances[groupName].telemetryObjectLevel;
+}
+/**
+ * Returns telemetry property of telemetry object
+ * @param groupName Group name to search for in the specified json data
+ * @param propertyName Property name that will be used to access and return the associated value
+ * @param jsonFilePath Optional path to the json config file
+ */
+export function readTelemetryObjectProperty(groupName: string, propertyName: string, jsonFilePath = telemetryJsonFilePath): string {
+    const jsonData = readTelemetryJsonData(jsonFilePath);
+    return jsonData.telemetryInstances[groupName][propertyName];
+}
+/**
+ * Writes to telemetry config file either appending to already existing file or creating new file
+ * @param groupName Group name of telemetry object
+ * @param telemetryLevel Whether user is sending basic or verbose telemetry
+ * @param jsonFilePath Optional path to the json config file
+ */
+
+export function writeTelemetryJsonData(groupName: string, telemetryObjectLevel: string, jsonFilePath = telemetryJsonFilePath): void {
+    if (fs.existsSync(jsonFilePath) && fs.readFileSync(jsonFilePath, "utf8") !== "") {
+        const telemetryJsonData = readTelemetryJsonData(jsonFilePath);
+        if (!groupNameExists(telemetryJsonData, groupName)) {
+        telemetryJsonData.telemetryInstances[groupName] = {telemetryObjectLevel: String};
+        telemetryJsonData.telemetryInstances[groupName].telemetryObjectLevel = telemetryObjectLevel;
+        fs.writeFileSync(jsonFilePath, JSON.stringify((telemetryJsonData), null, 2));
```

Comment:
I think there's a bug here in that if the group name does exist, we have no way to modify the telemetry level. There needs to be an else statement for the case where the group name does exist